dotCMS/src/main/java/com/dotmarketing/portlets/workflows/ajax/WfStepAjax.java

```diff
@@ -81,23 +82,56 @@ public void reorder(HttpServletRequest request, HttpServletResponse response) th
 		}
 			
 	}
-	
-	public void delete(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-		String stepId = request.getParameter("stepId");
 
-		WorkflowAPI wapi = APILocator.getWorkflowAPI();
-		
+	public void delete(final HttpServletRequest request,
+					   final HttpServletResponse response) throws ServletException, IOException {
+
+		final String stepId = request.getParameter("stepId");
+
 		try {
 
-			WorkflowStep step = wapi.findStep(stepId);
-			wapi.deleteStep(step);
-		} catch (DotDataException e) {
+			this.workflowHelper.deleteStep (stepId);
+		} catch (Exception e) {
 			Logger.error(this.getClass(),e.getMessage());
 			writeError(response, "</br> Delete Failed : </br>"+  e.getMessage());
 		}
-		
-		
+	} // delete.
+
+	/**
+	 * Associated an existing action to the step.
+	 * If the action or step does not exists, returns fail.
+	 * If the action is already associated to the step returns fail.
+	 * If the user does not have permission to do the action returns fail.
+	 * @param request   HttpServletRequest
+	 * @param response HttpServletResponse
+	 * @throws ServletException
+	 * @throws IOException
+	 */
+	public void addActionToStep(final HttpServletRequest request,
+								final HttpServletResponse response) throws ServletException, IOException {
+
+		final String stepId   = request.getParameter("stepId");
+		final String actionId = request.getParameter("actionId");
+
+		try {
+
+			final User user   = this.userWebAPI.getUser(request);
+
+			Logger.debug(this, "Adding the action: " + actionId +
+							", to the step: " + stepId);
+			this.workflowHelper.saveActionToStep (
+					new WorkflowActionStepBean.Builder()
+							.stepId(stepId)
+							.actionId(actionId)
+							.build(), user);
+
+			writeSuccess(response, stepId );
+		} catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)